test/Microsoft.AspNetCore.Authentication.Test/Google/GoogleMiddlewareTests.cs

```diff
@@ -781,6 +789,13 @@ private static TestServer CreateServer(GoogleOptions options, Func<HttpContext,
                         {
                             await context.Authentication.ChallengeAsync("Google");
                         }
+                        else if (req.Path == new PathString("/tokens"))
+                        {
+                            var authContext = new AuthenticateContext(TestExtensions.CookieAuthenticationScheme);
```

Comment:
Yeah I agree, basically its the internal guts exposed, which is why it feels a bit gross...